src/storage/hummock_sdk/src/table_watermark.rs

```diff
@@ -93,7 +92,8 @@ impl TableWatermarksIndex {
         self.index.get(&vnode).and_then(|epoch_watermarks| {
             epoch_watermarks
                 .upper_bound(Included(&epoch))
-                .value()
+                .peek_next()
+                .map(|(_, v)| v)
```

Comment:
This is very suspicious. `upper_bound` "returns a Cursor pointing at the last gap below the given bound," which means if we have `[1, 2, 3, 4]`, `upper_bound(Included(3))` will return the gap between `3` and `4`, then `peek_next` will return `4`, however previously we would get `3`.